Package.swift

```diff
@@ -1,47 +1,35 @@
-// swift-tools-version:5.2
+// swift-tools-version:5.3
 import PackageDescription
 
+#if os(macOS) && compiler(>=5.5) || compiler(>=5.6)
+let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50600.1")
+#elseif compiler(>=5.5)
+let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50500.0")
+#elseif compiler(>=5.4)
+let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50400.0")
+#elseif compiler(>=5.3)
+let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50300.0")
+#else
+fatalError("This version of Periphery does not support Swift <= 5.2.")
+#endif
+
 var dependencies: [Package.Dependency] = [
     .package(url: "https://github.com/apple/swift-system", from: "1.0.0"),
     .package(url: "https://github.com/jpsim/Yams", from: "4.0.0"),
     .package(url: "https://github.com/tadija/AEXML", from: "4.0.0"),
     .package(url: "https://github.com/apple/swift-argument-parser", from: "1.0.0"),
-    .package(name: "SwiftIndexStore", url: "https://github.com/kateinoigakukun/swift-indexstore", from: "0.0.0")
+    .package(name: "SwiftIndexStore", url: "https://github.com/kateinoigakukun/swift-indexstore", from: "0.0.0"),
+    .package(name: "SwiftSyntax", url: "https://github.com/apple/swift-syntax", swiftSyntaxVersion)
 ]
-#if swift(>=5.6)
-dependencies.append(
-    .package(
-        name: "SwiftSyntax",
-        url: "https://github.com/apple/swift-syntax",
-        .exact("0.50600.1")
-    )
-)
-#elseif swift(>=5.5)
-dependencies.append(
-    .package(
-        name: "SwiftSyntax",
-        url: "https://github.com/apple/swift-syntax",
-        .exact("0.50500.0")
-    )
-)
-#elseif swift(>=5.4)
-dependencies.append(
-    .package(
-        name: "SwiftSyntax",
-        url: "https://github.com/apple/swift-syntax",
-        .exact("0.50400.0")
-    )
-)
-#elseif swift(>=5.3)
+
+#if os(macOS) && compiler(>=5.5)
 dependencies.append(
     .package(
-        name: "SwiftSyntax",
-        url: "https://github.com/apple/swift-syntax",
-        .exact("0.50300.0")
+        name: "StaticSwiftSyntaxParser",
+        url: "https://gist.github.com/liamnichols/92f8fdcf2864d0fd1619a18828acafb8.git",
+        .branch("main")
```

Comment:
Ok, so recap on this.. I've had to move this out of Package.swift into a dedicated dependency because using `.binaryTarget` in Package.swift currently breaks `swift package describe`, which the tests rely on. [SR-15243](https://bugs.swift.org/browse/SR-15243) (and [SR-15065](https://bugs.swift.org/browse/SR-15065) when working around things with a local binary target).

If you'd be happy with this change and want to proceed, I'd recommend that you push the gist up to it's own repository under the peripheryapp org, then tag it with something like `5.6` so we can do this:

```suggestion
        name: "StaticSwiftSyntaxParser",
        url: "https://github.com/peripheryapp/static-swift-syntax-parser.git",
        .exact("5.6")
```

---

Another alternative that I had considered was to just fork https://github.com/apple/swift-syntax entirely and branch off 0.50600.1 and tweak the original **Package.swift** so that it uses [StaticInternalSwiftSyntaxParser](http://github.com/keith/StaticInternalSwiftSyntaxParser) instead of the dylib. We'd then be able to update line 5 to something like this:

```diff
  #if os(macOS) && compiler(>=5.5) || compiler(>=5.6)
- let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50600.1")
+ let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50600.1-static")
  #elseif compiler(>=5.5)
  let swiftSyntaxVersion: Package.Dependency.Requirement = .exact("0.50500.0")
```

As well as a few other tweaks

Reply:
I tried this out in #480